sunpy/lightcurve/lightcurve.py

```diff
@@ -380,6 +380,37 @@ def time_range(self):
         object"""
         return TimeRange(self.data.index[0], self.data.index[-1])
 
+    def concatenate(self, otherlightcurve):
+        """Concatenate another light curve. This function will check and remove
+        any duplicate times.
+
+        Parameters
+        ----------
+        otherlightcurve : `~sunpy.lightcurve.LightCurve`
+            Another lightcurve of the same type.
+
+        Returns
+        -------
+        newlc : `~sunpy.lightcurve.LightCurve`
+            A new lightcurve.
+        """
+        if isinstance(otherlightcurve, self.__class__):
```

Comment:
You can remove the branch when the classes are the same by directly testing for when they are not the same:
if not isinstance(otherlightcurve, self.__class__):
    raise TypeError("Lightcurve classes must match.")